core/domain/email_services.py

```
@@ -305,9 +305,9 @@ def convert_email_to_loggable_string(
         """ % (
         ', '.join(filenames) if len(filenames) > 0 else 'None'
     )
-    loggable_msg = (
-        textwrap.dedent(msg)
-        + textwrap.dedent(optional_msg_description)
-        + textwrap.dedent(attachments_msg_description)
+    loggable_msg = '%s%s%s' % (
+        textwrap.dedent(msg),
+        textwrap.dedent(optional_msg_description),
+        textwrap.dedent(attachments_msg_description),
     )
```

Ruff (0.15.21)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)